cvat/apps/engine/task.py

```diff
@@ -272,13 +272,17 @@ def rq_handler(job, exc_type, exc_value, traceback):
 ############################# Internal implementation for server API
 
 class _FrameExtractor:
-    def __init__(self, source_path, compress_quality, flip_flag=False):
+    def __init__(self, source_path, compress_quality, flip_flag=False, frame_rate=0):
```

Comment:
The main problem with the approach is you will get annotation file which doesn't correspond to the video file. For example, originally you have a video file with 30FPS. If someone annotates the whole video file we will get `<frame000000>, <frame000001>, ...` inside the annotation file for every frame. Now you say that we can provide "custom" FPS. You will get another `<frame000000>, <frame000001>, ...` which don't correspond to the original video file. How to much if you don't store extra parameters (e.g. custom frame rate) inside the annotation file?

Reply:
I'm not familiar with the internals of backend. Is there any difference between upload a video directly with upload all frame images pre-extracted locally?

Reply:
@zliang7, if you still have questions it is better to organize a meeting and discuss.